sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -0,0 +1,135 @@
+import os
+import urllib
+
+from azure.core.pipeline.policies import HttpLoggingPolicy
+import logging
+import types
+
+
+
+class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
+
+    def __init__(self, logger=None, enable_diagnostics_logging=False, **kwargs):
+        self._diagnostics_mode = enable_diagnostics_logging
+        if not enable_diagnostics_logging:
+            super().__init__(logger, **kwargs)
+        else:
+            self.logger = logger or logging.getLogger(__name__)
+            self.allowed_query_params = set()
+
+    def on_request(self, request):  # type: (PipelineRequest) -> None
+        if self._diagnostics_mode:
+            http_request = request.http_request
```

Comment:
Most of this logging looks like it's using the existing implementation, which means maintaining lots of duplicate code.
I would recommend something like:
```
def on_request(self, request):
    super().on_request(request)
    diagnostics_logging = request.context.setdefault(
        "enable_diagnostics",
        options.pop("enable_diagnostics, self._enable_diagnostics)
    )
    if diagnostics_logging and logger.isEnabledFor(logging.INFO):
        # Add the Cosmos specific stuff here
```

Reply:
Given the logic of the code, would it make sense to do this? The diagnostics logging includes additional information for headers that the original implementation excludes, then there's also the multirecord option logic.